cnijfilter2: 6.10 -> 6.40

N-Parsons opened the pull request:
###### Description of changes

Update cnijfilter2 to the latest 6.40 package to support newer printers. Canon have added a new binary and renamed a folder, but no changes are breaking.

I've tested this package on my Canon Pixma G650 (which requires >=6.30) and all seems to be working well.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

N-Parsons committed:
cnijfilter2: 6.10 -> 6.40

maxeaubrey commented:
cnijfilter2 no longer builds on top of master. The same is true for the version that's already there.

```
/nix/store/0gzd0049bywqhzi5anbydsxil7r1lfrj-binutils-2.38/bin/ld: cnijifnet2.o:(.bss+0x0): multiple definition of `GET_PROTOCOL'; cnijlgmon3.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:404: cnijlgmon3] Error 1
make[2]: Leaving directory '/build/source/lgmon3/src'
make[1]: *** [Makefile:416: all-recursive] Error 1
make[1]: Leaving directory '/build/source/lgmon3'
make: *** [Makefile:348: all] Error 2
error: build of '/nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv' on 'ssh://remote-build-user@sintra.home.arpa' failed: builder for '/nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv' failed with exit code 2;
       last 10 log lines:
       > mv -f .deps/cnijutil.Tpo .deps/cnijutil.Po
       > /nix/store/qm2lv1gpbyn0rsfai40cbvj3h4gz69yc-bash-5.1-p16/bin/bash ../libtool  --tag=CC   --mode=link gcc  -O2 -Wall -I/nix/store/ngz3ki94j4rkgldqnwpsascm3hfmd9xn-libusb-1.0.26-dev/include/libusb-1.0 -I./common -I/nix/store/blmrii1hvd4cc3lralj91ll7a079pxd7-libxml2-2.9.14-dev/include/libxml2   -o cnijlgmon3 cnijcomif.o cnijifusb.o cnijlgmon3.o cnijifnet.o keytext.o cnijifnet2.o cnijutil.o -ldl -lusb-1.0 -lcnnet2 -lxml2 -lcnbpcnclapicom2 -lcnbpnet30 -lcups
       > libtool: link: gcc -O2 -Wall -I/nix/store/ngz3ki94j4rkgldqnwpsascm3hfmd9xn-libusb-1.0.26-dev/include/libusb-1.0 -I./common -I/nix/store/blmrii1hvd4cc3lralj91ll7a079pxd7-libxml2-2.9.14-dev/include/libxml2 -o cnijlgmon3 cnijcomif.o cnijifusb.o cnijlgmon3.o cnijifnet.o keytext.o cnijifnet2.o cnijutil.o  -ldl -lusb-1.0 -lcnnet2 -lxml2 -lcnbpcnclapicom2 -lcnbpnet30 -lcups
       > /nix/store/0gzd0049bywqhzi5anbydsxil7r1lfrj-binutils-2.38/bin/ld: cnijifnet2.o:(.bss+0x0): multiple definition of `GET_PROTOCOL'; cnijlgmon3.o:(.bss+0x0): first defined here
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [Makefile:404: cnijlgmon3] Error 1
       > make[2]: Leaving directory '/build/source/lgmon3/src'
       > make[1]: *** [Makefile:416: all-recursive] Error 1
       > make[1]: Leaving directory '/build/source/lgmon3'
       > make: *** [Makefile:348: all] Error 2
       For full logs, run 'nix log /nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv'.
error: builder for '/nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv' failed with exit code 1;
       last 10 log lines:
       > mv -f .deps/cnijutil.Tpo .deps/cnijutil.Po
       > /nix/store/qm2lv1gpbyn0rsfai40cbvj3h4gz69yc-bash-5.1-p16/bin/bash ../libtool  --tag=CC   --mode=link gcc  -O2 -Wall -I/nix/store/ngz3ki94j4rkgldqnwpsascm3hfmd9xn-libusb-1.0.26-dev/include/libusb-1.0 -I./common -I/nix/store/blmrii1hvd4cc3lralj91ll7a079pxd7-libxml2-2.9.14-dev/include/libxml2   -o cnijlgmon3 cnijcomif.o cnijifusb.o cnijlgmon3.o cnijifnet.o keytext.o cnijifnet2.o cnijutil.o -ldl -lusb-1.0 -lcnnet2 -lxml2 -lcnbpcnclapicom2 -lcnbpnet30 -lcups
       > libtool: link: gcc -O2 -Wall -I/nix/store/ngz3ki94j4rkgldqnwpsascm3hfmd9xn-libusb-1.0.26-dev/include/libusb-1.0 -I./common -I/nix/store/blmrii1hvd4cc3lralj91ll7a079pxd7-libxml2-2.9.14-dev/include/libxml2 -o cnijlgmon3 cnijcomif.o cnijifusb.o cnijlgmon3.o cnijifnet.o keytext.o cnijifnet2.o cnijutil.o  -ldl -lusb-1.0 -lcnnet2 -lxml2 -lcnbpcnclapicom2 -lcnbpnet30 -lcups
       > /nix/store/0gzd0049bywqhzi5anbydsxil7r1lfrj-binutils-2.38/bin/ld: cnijifnet2.o:(.bss+0x0): multiple definition of `GET_PROTOCOL'; cnijlgmon3.o:(.bss+0x0): first defined here
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [Makefile:404: cnijlgmon3] Error 1
       > make[2]: Leaving directory '/build/source/lgmon3/src'
       > make[1]: *** [Makefile:416: all-recursive] Error 1
       > make[1]: Leaving directory '/build/source/lgmon3'
       > make: *** [Makefile:348: all] Error 2
       For full logs, run 'nix log /nix/store/y3zx9v50p08hpkfrr5x59rck6hnaaamh-cnijfilter2-6.40.drv'.
```

N-Parsons committed:
Fix compilation error related to duplicate definitions of GET_PROTOCOL

- Resolves the compilation issue reported at https://github.com/NixOS/nixpkgs/pull/180681#issuecomment-1192304711
- An identical issue was previously reported in Gentoo: https://bugs.gentoo.org/723186
- This patch is taken from the solution of Alfredo Tupone (https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=24688d64544b43f2c14be54531ad8764419dde09)

N-Parsons commented:
@maxeaubrey, it looks like Gentoo had the same issue a while back: https://bugs.gentoo.org/723186

I've added [the patch used by Gentoo](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=24688d64544b43f2c14be54531ad8764419dde09) and the package compiles again.
